plugins/CoreAdminHome/Tasks.php

```diff
@@ -258,6 +262,48 @@ public function updateSpammerBlacklist()
         Option::set(ReferrerSpamFilter::OPTION_STORAGE_NAME, serialize($list));
     }
 
+    public function purgeOrphanedArchives()
+    {
+        $segmentHashesByIdSite = $this->getSegmentHashesByIdSite();
+        $archiveTables = ArchiveTableCreator::getTablesArchivesInstalled('numeric');
+
+        $datesPurged = array();
+        foreach ($archiveTables as $table) {
+            $date = ArchiveTableCreator::getDateFromTableName($table);
+            list($year, $month) = explode('_', $date);
+
+            $dateObj = Date::factory("$year-$month-15");
+
+            $this->archivePurger->purgeDeletedSiteArchives($dateObj);
+            $this->archivePurger->purgeDeletedSegmentArchives($dateObj, $segmentHashesByIdSite);
+
+            $datesPurged[$date] = true;
+        }
+    }
+
+    /**
+     * Get a list of all segment hashes that currently exist, indexed by idSite.
+     * @return array
+     */
+    public function getSegmentHashesByIdSite()
+    {
+        //Get a list of hashes of all segments that exist now
+        $sql = "SELECT DISTINCT definition, enable_only_idsite FROM " . Common::prefixTable('segment')
+            . " WHERE deleted = 0";
+        $rows = Db::fetchAll($sql);
+        $segmentHashes = array();
+        foreach ($rows as $row) {
+            try {
+                $segment = new Segment($row['definition'], array());
+                $segmentHashes[(int)$row['enable_only_idsite']][] = $segment->getHash();
+            } catch (\Exception $ex) {
+                //Segment is invalid ... into the sea its archives shall go
```

Comment:
be good to log a notice in this case maybe? This will help us reproduce why potentially archives were deleted. 

Actually, I'm thinking it may be too risky if we delete them. There could be edge cases where we eg deploy code or there are temporary bugs where a segment can not be found. In this case we wouldn't want to delete the archives as it could cause quite a bit of trouble, especially on the cloud maybe? 

I wonder if we should in this case maybe try to get the segment hash manually by just doing `md5(urldecode($row['definition']));`. Maybe we could actually do this in general... then we also don't need to care about passing `idSites` to `new Segment()` etc? We would just need to create a static method for it that we use here like `Segment::getSegmentHash()` which could be also used in `$segment->getHash`?

Reply:
Have extracted into a separate method as suggested, now skipping segment constructor/validation and just calculating the hash on the definition string